src/main/java/br/com/caelum/viagens/voos/controller/dto/input/NewParadaDaRotaInputDto.java

```diff
@@ -0,0 +1,40 @@
+package br.com.caelum.viagens.voos.controller.dto.input;
+
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Positive;
+
+import br.com.caelum.viagens.voos.model.Parada;
+import br.com.caelum.viagens.voos.model.TipoParada;
+import br.com.caelum.viagens.voos.validator.annotations.TipoParadaExistente;
+
+public class NewParadaDaRotaInputDto {
+	
+	@TipoParadaExistente
+	@NotNull
+	private String tipo;
```

Comment:
Ponto positivo! Você criou uma annotation seguindo o necessário da BeanValidation e imagino que tenha ficado ainda mais claro como funciona o esquema de validação. 

Ponto de atenção:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*
+
*Avaliamos de forma lógica o nível de complicação de cada trecho de código. Aqui não tem espaço para feeling. Usamos a teoria da carga cognitiva e medimos o nível de pontos de entendimentos necessário por trecho.*

Se você cria a Enum, você trava as possibilidades em tempo de compilação usando uma coisa pronta da linguagem. E tudo que é pronto da linguagem tende a ser mais fácil de ser entendido.

Reply:
ok, por enquanto eu vou manter e mais para frente vou criar um handler para tratar erros de conversão, pode ser?